frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts

```diff
@@ -246,4 +247,86 @@ export class VirtualMachine extends KubevirtDetailView {
       await this.waitForStatus(VM_STATUS.Off, VM_IMPORT_TIMEOUT_SECS);
     }
   }
+
+  async import({
+    provider,
+    instanceConfig,
+    sourceVMName,
+    name,
+    description,
+    operatingSystem,
+    flavorConfig,
+    workloadProfile,
+    storageResources,
+    networkResources,
+    cloudInit,
+  }: VMImportConfig) {
+    const importWizard = new ImportWizard();
+    await this.navigateToListView();
+    await importWizard.openWizard();
+
+    // General section
+    await importWizard.selectProvider(provider);
+    await importWizard.waitForVMWarePod();
+    await importWizard.configureInstance(instanceConfig);
+
+    await importWizard.connectToInstance();
+    await importWizard.waitForInstanceSync();
+
+    await importWizard.selectSourceVirtualMachine(sourceVMName);
+    await importWizard.waitForInstanceSync();
+
+    if (operatingSystem) {
+      await importWizard.selectOperatingSystem(operatingSystem as string);
+    }
+    if (flavorConfig) {
+      await importWizard.configureFlavor(flavorConfig);
+    }
+    if (workloadProfile) {
+      await importWizard.selectWorkloadProfile(workloadProfile);
+    }
+    if (name) {
+      await importWizard.fillName(name);
+    }
+    if (description) {
+      await importWizard.fillDescription(description);
+    }
+    await importWizard.next();
+
+    // Networking
+    // Frst update imported network interfaces to comply with k8s
+    await importWizard.updateImportedNICs();
+    // Optionally add new interfaces, if any
+    if (networkResources !== undefined) {
+      for (const NIC of networkResources) {
+        await importWizard.addNIC(NIC);
+      }
+    }
+    await importWizard.next();
+
+    // Storage
+    // First update disks that come from the source VM
+    await importWizard.updateImportedDisks();
+    // Optionally add new disks
+    if (networkResources !== undefined) {
+      for (const disk of storageResources) {
+        await importWizard.addDisk(disk);
+      }
+    }
+    await importWizard.next();
+
+    // Advanced - Cloud Init
+    if (cloudInit !== undefined) {
+      await importWizard.configureCloudInit(cloudInit);
+    }
+    await importWizard.next();
+    // Advanced - Virtual HW
+    await importWizard.next();
+    // Review
+    await importWizard.confirmAndCreate();
+    await importWizard.waitForCreation();
```

Comment:
Would it make sense to check the result?

Reply:
I don't think additional checking errors would be beneficial at the moment.
Currently the function call `await importWizard.waitForCreation();` waits for success message, so if the creation fails for some reason, exception will be raised here and we'll have a screenshot taken wit the error.

We could improve it by reading the error message and returning it if VM creation fails, so that we can work with the error in the test function. This could be definitely useful for negative test scenarios.

I would leave it for a separate effort though.